test/Dialect/FIRRTL/expand-whens.mlir

```diff
@@ -56,9 +56,9 @@ firrtl.module @simulation(in %clock : !firrtl.clock, in %p : !firrtl.uint<1>, in
 // CHECK-NEXT:   %1 = firrtl.and %p, %enable : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
 // CHECK-NEXT:   firrtl.stop %clock, %1, 0
 // CHECK-NEXT:   %2 = firrtl.and %p, %enable : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
-// CHECK-NEXT:   firrtl.assert %clock, %p, %2, ""
+// CHECK-NEXT:   firrtl.assert %clock, %p, %2, "" {eventControl = 0 : i32, isConcurrent = false}
```

Comment:
Can we still omit these, or was it intentional to change this behavior?

Switching to `DefaultValuedOptionalAttr` does the trick here, and I think it helps readability to omit.

OTOH https://github.com/llvm/llvm-project/commit/7d273fde110d7735512c3b71a83eb88e89d189cc suggests the use of DefaultValuedAttr may be more python-friendly (?) and otherwise avoids per-access checks. and even using `DefaultValuedOptionalAttr` some things like the verbatim symbols array (cc #2088) are printed if set/empty (but omitted in round-trip where they weren't specified), which I haven't looked into much (maybe default value needs to be empty array vs null attr?).

Reply:
It seemed like a good idea to make the attributes non-optional, the idea being the verification constructs must have the `eventControl` and `isConcurrent`. I agree, it helps readability to omit them though.
I think we need to handle this case by case. I will open an issue to track this.

Reply:
Awesome, thanks! :+1: